reconcile/utils/jira_client.py

```diff
@@ -15,21 +18,41 @@ def __init__(self, jira_board, settings=None):
         token_auth = self.secret_reader.read(token)
         self.jira = JIRA(self.server, token_auth=token_auth)
 
-    def get_issues(self, fields=None):
+    def get_issues(self, fields: Optional[Mapping] = None) -> list:
         block_size = 100
         block_num = 0
-        all_issues = []
+
+        all_issues = []  # type: ignore
```

Comment:
Just curious why this needs to be ignored?

Reply:
mypy is pretty unespecific: 
```
reconcile/utils/jira_client.py:25: error: Need type annotation for "all_issues" (hint: "all_issues: List[<type>] = ...")
```
(and if I try to cast it to `list[Issue]`  I get a suggestion for something truly awful)

Reply:
That's the expected behavior for collections: https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#explicit-types-for-collections

I added `list[Issue]` and see:

```
reconcile/utils/jira_client.py:32: error: Argument 4 to "search_issues" of "JIRA" has incompatible type "**Dict[str, str]"; expected "bool"
reconcile/utils/jira_client.py:33: error: Argument 1 to "extend" of "list" has incompatible type "Union[List[Dict[str, Any]], ResultList[Issue]]"; expected "Iterable[Issue]"
```

I think the first issue is valid. You didn't add a type to `kwargs`, so `dict[str, str]` is inferred because of `kwargs['fields'] = ','.join(fields)`. If you change this to `kwargs: dict[str, Any]`, then that error goes away.

The remaining issue is that `search_issues` return type is not `Issue`, it's: `-> Union[List[Dict[str, Any]], ResultList[Issue]]`.

This can be fixed by changing `all_issues` to: `all_issues: list[Union[dict, Issue]] = []` and the return type to the same.

It takes a little bit of time to run these down, but Mypy is doing the right thing here. Our expectations are not inline with what the Jira client is actually returning. If we're certain that `ResultList[Issue]` will always be returned, despite what the `search_issues` types say, then [this might be helpful](https://mypy.readthedocs.io/en/stable/common_issues.html).